File: app/src/main/java/com/sonymobile/agent/robot/camera/CvUtils.java
package com.sonymobile.agent.robot.camera;

import android.graphics.Bitmap;
import android.util.Log;

import java.nio.ByteBuffer;

public final class CvUtils {

    static {
        try {
            System.loadLibrary("cvUtils");
            Log.d("yama cvUtils","LibraryFound");
        } catch (UnsatisfiedLinkError noLibraryFound) {
            Log.e("yama cvUtils","noLibraryFound");
            // Do nothing
        }
    }

    public static boolean nativeYuvCropRotateToRgb(byte[] src_yuv, boolean isNV21, int src_width, int src_height, int[] crop_area, int rotate, ByteBuffer dst, int dst_width, int dst_height, int dst_ch){
        return yuvCropRotateToRgb(src_yuv,isNV21,src_width,src_height,src_height,crop_area,rotate,dst,dst_width,dst_height,dst_ch);
    }
    public static boolean nativeYuvToRgb(byte[] src_yuv,boolean isNV21,int src_width,int src_height,ByteBuffer dst,int dst_width,int dst_height,int dst_ch){
        return yuvToRgb(src_yuv,isNV21,src_width,src_height,src_height,dst,dst_width,dst_height,dst_ch);
    }
    public static boolean nativeYuvToBitmap(byte[] src_yuv,boolean isNV21,int src_width,int src_height,int src_pitch,Bitmap bitmap){
        return yuvToBitmap(src_yuv,isNV21,src_width,src_height,src_pitch,bitmap);
    }


    private static native boolean yuvCropRotateToRgb(byte[] src_yuv,boolean isNV21,int src_width,int src_height,int src_pitch,int[] crop_area,int rotate,ByteBuffer dst,int dst_width,int dst_height,int dst_ch);
    private static native boolean yuvToRgb(byte[] src_yuv,boolean isNV21,int src_width,int src_height,int src_pitch,ByteBuffer dst,int dst_width,int dst_height,int dst_ch);
    private static native boolean yuvToBitmap(byte[] src_yuv,boolean isNV21,int src_width,int src_height,int src_pitch,Bitmap bitmap);
}

File: app/src/main/java/com/sonymobile/agent/robot/camera/CvUtils.kt
package com.sonymobile.agent.robot.camera

import android.graphics.Bitmap
import android.graphics.Rect
import android.util.Log
import java.nio.ByteBuffer

class CvUtils{

    companion object {
        init {
            try {
                System.loadLibrary("cvUtils")
                Log.d("yama cvUtils", "LibraryInit")
            } catch (noLibraryFound: UnsatisfiedLinkError) {
                Log.e("yama cvUtils", "noLibraryNotFound")
            }
        }
        const val YUV_I420=0
        const val YUV_NV12=1
        const val YUV_NV21=2
    }


    external fun yuvCropRotateToRgb(src_yuv: ByteArray, yuv_type: Int, src_width: Int, src_height: Int, src_pitch: Int, crop_area: IntArray, rotate: Int, dst: ByteBuffer, dst_width: Int, dst_height: Int, dst_ch: Int): Boolean
    external fun yuvToRgb(src_yuv: ByteArray, yuv_type: Int, src_width: Int, src_height: Int, src_pitch: Int, dst: ByteBuffer, dst_width: Int, dst_height: Int, dst_ch: Int): Boolean
    external fun yuvToBitmap(src_yuv: ByteArray,yuv_type:Int,src_wisth:Int,src_height:Int,src_pitch:Int,bitmap: Bitmap,dst_width:Int=src_wisth,dst_height:Int=src_height):Boolean
    external fun getMotionArea(img1: ByteBuffer, img2: ByteBuffer, width: Int, height: Int, rect: IntArray): Boolean
}
